### PR TITLE
URLFile: default value test

### PR DESCRIPTION
**Pull request by gregjhogan**
configuration expected by pipeline

### CODE REVIEW

File: tools/lib/tests/test_caching.py [hunk 1]
```diff
@@ -34,6 +35,28 @@ def do_HEAD(self):
 
 class TestFileDownload(unittest.TestCase):
 
+  def test_defaults(self):
+    # TODO: parameterize the defaults so we don't rely on hard-coded values in xx
+
+    self.assertEqual(URLFile.pool_manager().pools._maxsize, 10) # PoolManager num_pools param
```
Comment (adeebshihadeh) [inline tools/lib/tests/test_caching.py:41]: can you add a comment that makes it clear that these defaults are for the pipeline? or maybe put it in the test name?

File: tools/lib/tests/test_caching.py [hunk 2]
```diff
@@ -34,6 +35,28 @@ def do_HEAD(self):
...
+
+    retry_defaults = {
+      "total": 5,
+      "backoff_factor": 0.5,
+      "status_forcelist": [409, 429, 503, 504],
+    }
+    for k, v in retry_defaults.items():
+      self.assertEqual(getattr(URLFile.pool_manager().connection_pool_kw["retries"], k), v)
+
+     # caching off by default
```
Comment (adeebshihadeh) [inline tools/lib/tests/test_caching.py:61]: extra space

File: tools/lib/tests/test_caching.py [hunk 3]
```diff
@@ -34,6 +35,28 @@ def do_HEAD(self):
...
+    self.assertEqual(URLFile("http://fake.url")._force_download, True)
```
Comment (adeebshihadeh) [inline tools/lib/tests/test_caching.py:62]: this is both a brittle test and doesn't directly test what you want. checking that the file doesn't end up in `Paths.download_cache_root()` would be a better test, but ideally we check if anything was written to disk

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

File: tools/lib/tests/test_caching.py [hunk 4]
```diff
@@ -34,6 +35,32 @@ def do_HEAD(self):
...
+      "backoff_factor": 0.5,
+      "status_forcelist": [409, 429, 503, 504],
+    }
+    for k, v in retry_defaults.items():
+      self.assertEqual(getattr(URLFile.pool_manager().connection_pool_kw["retries"], k), v)
+
+    try:
+      del os.environ["FILEREADER_CACHE"]
+    except KeyError:
+      pass
```
Comment (adeebshihadeh) [inline tools/lib/tests/test_caching.py:60]: not sure why this is set. our tests should always have a clean environment. @jnewb1 you know what's going on here?

if we have to do this, this is cleaner
```suggestion
    os.environ.pop("FILEREADER_CACHE", None)
```

Comment (jnewb1) [reply tools/lib/tests/test_caching.py:60]: [this](https://github.com/commaai/openpilot/blob/master/conftest.py#L29-L34) just ensures that any environment variables changed within a single test don't get propagated to the next test. I think pytest uses some environment variables so I don't think we can clear them totally, but there might be a better way to handle this

Comment (jnewb1) [reply tools/lib/tests/test_caching.py:60]: FILEREADER_CACHE is also set in CI:
https://github.com/commaai/openpilot/blob/master/.github/workflows/selfdrive_tests.yaml#L22

**Pull request merged**